From ebb9aebd77c918e37167ad1df95fccce600943cc Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Wed, 2 May 2018 16:23:21 +0200
Subject: [PATCH 1/4] conf/ucm: Add a UCM profile for Dell WD15 Dock USB-audio

USB-audio device on Dell WD15 docking station provides two individual
PCM streams, one for headphone and another for line out.  A UCM
profile gives the proper roles for these.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
(cherry picked from commit 8ebb40c96970c913719a75deb2fe82ba2e257386)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
---
 .../ucm/Dell-WD15-Dock/Dell-WD15-Dock.conf    |  5 ++++
 src/conf/ucm/Dell-WD15-Dock/HiFi.conf         | 26 +++++++++++++++++++
 src/conf/ucm/Dell-WD15-Dock/Makefile.am       |  4 +++
 src/conf/ucm/Makefile.am                      |  2 +-
 4 files changed, 36 insertions(+), 1 deletion(-)
 create mode 100644 src/conf/ucm/Dell-WD15-Dock/Dell-WD15-Dock.conf
 create mode 100644 src/conf/ucm/Dell-WD15-Dock/HiFi.conf
 create mode 100644 src/conf/ucm/Dell-WD15-Dock/Makefile.am

diff --git a/src/conf/ucm/Dell-WD15-Dock/Dell-WD15-Dock.conf b/src/conf/ucm/Dell-WD15-Dock/Dell-WD15-Dock.conf
new file mode 100644
index 00000000..f4f0a8d9
--- /dev/null
+++ b/src/conf/ucm/Dell-WD15-Dock/Dell-WD15-Dock.conf
@@ -0,0 +1,5 @@
+Comment "USB-audio on Dell WD15 docking station"
+SectionUseCase."HiFi" {
+	File "HiFi.conf"
+	Comment "Default"
+}
diff --git a/src/conf/ucm/Dell-WD15-Dock/HiFi.conf b/src/conf/ucm/Dell-WD15-Dock/HiFi.conf
new file mode 100644
index 00000000..b07f3440
--- /dev/null
+++ b/src/conf/ucm/Dell-WD15-Dock/HiFi.conf
@@ -0,0 +1,26 @@
+SectionDevice."Headphone" {
+	Comment "Headphone"
+
+	Value {
+		PlaybackChannels "2"
+		PlaybackPCM "hw:WD15Dock,0"
+	}
+}
+
+SectionDevice."LineOut" {
+	Comment "Line Out"
+
+	Value {
+		PlaybackChannels "2"
+		PlaybackPCM "hw:WD15Dock,1"
+	}
+}
+
+SectionDevice."Mic" {
+	Comment "Microphone"
+
+	Value {
+		CaptureChannels "2"
+		CapturePCM "hw:WD15Dock,0"
+	}
+}
diff --git a/src/conf/ucm/Dell-WD15-Dock/Makefile.am b/src/conf/ucm/Dell-WD15-Dock/Makefile.am
new file mode 100644
index 00000000..7ab58730
--- /dev/null
+++ b/src/conf/ucm/Dell-WD15-Dock/Makefile.am
@@ -0,0 +1,4 @@
+alsaconfigdir = @ALSA_CONFIG_DIR@
+ucmdir = $(alsaconfigdir)/ucm/Dell-WD15-Dock
+ucm_DATA = Dell-WD15-Dock.conf HiFi.conf
+EXTRA_DIST = $(ucm_DATA)
diff --git a/src/conf/ucm/Makefile.am b/src/conf/ucm/Makefile.am
index 61870f87..6e3713d0 100644
--- a/src/conf/ucm/Makefile.am
+++ b/src/conf/ucm/Makefile.am
@@ -1 +1 @@
-SUBDIRS=DAISY-I2S PandaBoard PandaBoardES SDP4430 tegraalc5632 PAZ00 GoogleNyan HDAudio-Lenovo-DualCodecs broadwell-rt286 skylake-rt286 VEYRON-I2S chtrt5645 DB410c apq8064-tabla-snd-card Tuna tegra-rt5640 Manta-I2S Manta-SPDIF msm8974-taiko-mtp-snd-card
+SUBDIRS=DAISY-I2S PandaBoard PandaBoardES SDP4430 tegraalc5632 PAZ00 GoogleNyan HDAudio-Lenovo-DualCodecs broadwell-rt286 skylake-rt286 VEYRON-I2S chtrt5645 DB410c Dell-WD15-Dock apq8064-tabla-snd-card Tuna tegra-rt5640 Manta-I2S Manta-SPDIF msm8974-taiko-mtp-snd-card
-- 
2.17.1

